File: 2264-largest-3-same-digit-number-in-string/2264-largest-3-same-digit-number-in-string.py
class Solution:
    def largestGoodInteger(self, num: str) -> str:
        valids = ["999", "888", "777", "666", "555", "444", "333", "222", "111", "000"]
        
        for valid in valids:
            if valid in num:
                return valid

        return ""

File: 2264-largest-3-same-digit-number-in-string/2264-largest-3-same-digit-number-in-string.java
class Solution {
    public String largestGoodInteger(String num) {
        int max = -1;
        for (int i = 2; i < num.length(); i++) {
            if (num.charAt(i) == num.charAt(i-1) && num.charAt(i) == num.charAt(i-2)) {
                int number = num.charAt(i) - '0';
                max = Math.max(number, max);
            }
        }
        return max >= 0 ? ""+max+max+max : "";
    }
}